LeagueToolkit/IO/OBJ/OBJFile.cs

```diff
@@ -13,44 +14,147 @@ public class OBJFile
             this.Vertices = vertices;
             this.UVs = uvs;
             this.Normals = normals;
-            for (int i = 0; i < indices.Count; i += 3)
+            this.Groups.Add(new OBJGroup("default", "defaultMat", indices, indices, indices));
+        }
+
+        public OBJFile(List<Vector3> vertices, Dictionary<Tuple<string, string>, List<uint>> groups, List<Vector2> uvs, List<Vector3> normals, string mtlFile = "")
+        {
+            MaterialsFile = mtlFile;
+            this.Vertices = vertices;
+            this.UVs = uvs;
+            this.Normals = normals;
+
+            foreach (KeyValuePair<Tuple<string, string>, List<uint>> group in groups)
             {
-                uint[] faceIndices = new uint[] { indices[i], indices[i + 1], indices[i + 2] };
-                this.Faces.Add(new OBJFace(faceIndices, faceIndices, faceIndices));
+                this.Groups.Add(new OBJGroup(group.Key.Item1, group.Key.Item2, group.Value, group.Value, group.Value));
             }
         }
 
+        public OBJFile(List<OBJFile> objs)
+        {
+            uint indicesOffset = 0;
+            foreach (OBJFile obj in objs)
+            {
+                this.Comments.AddRange(obj.Comments);
+                this.Vertices.AddRange(obj.Vertices);
+                this.UVs.AddRange(obj.UVs);
+                this.Normals.AddRange(obj.Normals);
+
+                for (int i = 0; i < obj.Groups.Count; i++)
+                {
+                    OBJGroup group = obj.Groups[i];
+
+                    if (group.Faces.Count <= 0)
+                    {
+                        continue;
+                    }
+
+                    int mainGroupIndex = this.Groups.FindIndex(g => g.Name == group.Name);
+                    List<uint> tempIndices = new();
+
+                    if (mainGroupIndex != -1)
+                    {
+                        // Append vertex indices
+                        foreach (OBJFace face in group.Faces)
+                        {
+                            foreach (uint indice in face.VertexIndices)
+                            {
+                                tempIndices.Add(indice + indicesOffset);
+                            }
+                        }
+
+                        this.Groups[mainGroupIndex].Faces.AddRange(new OBJGroup(group.Name, group.Material, tempIndices, tempIndices, tempIndices).Faces);
+                    }
+                    else
+                    {
+                        // Append vertex indices
+                        foreach (OBJFace face in group.Faces)
+                        {
+                            foreach (uint indice in face.VertexIndices)
+                            {
+                                tempIndices.Add(indice + indicesOffset);
+                            }
+                        }
+
+                        this.Groups.Add(new OBJGroup(group.Name, group.Material, tempIndices, tempIndices, tempIndices));
+                    }
```

Comment:
Would probably read better if the if was put at the bottom to just switch the add/addrange call and the two foreach loops were merged together (they are the same, no?).